lib/player/drm_scheme_info.js

```diff
@@ -111,6 +110,35 @@ shaka.player.DrmSchemeInfo.Restrictions = function() {
    * @expose
    */
   this.maxWidth = null;
+
+  /**
+   * If set, specifies a maximum bandwidth for video tracks.
+   *
+   * @type {?number}
+   * @expose
+   */
+  this.maxBandwidth = null;
+
+  /**
+   * If set, specifies a minimum bandwidth for video tracks.
+   *
+   * @type {?number}
+   * @expose
+   */
+  this.minBandwidth = null;
+};
+
+
+/**
+ * @return {!shaka.player.DrmSchemeInfo.Restrictions}
+ *     A copy of the current restrictions
+ */
+shaka.player.DrmSchemeInfo.Restrictions.prototype.clone = function() {
+  var restrictions = new shaka.player.DrmSchemeInfo.Restrictions();
+  for (var key in this) {
```

Comment:
Please put 'struct' back on Restrictions.  I really think it's important to have the compiler check that the correct member names are being used.

To make this work, I see two choices:

1. Have clone() say "restrictions.x = this.x" explicitly for all 4 properties.  It's duplication, but at least it's maintained right next to the constructor, so updates should be easy to keep in sync.
2. Have the constructor take an optional Restrictions instance as an argument, such that the member declaration is of the form: "this.x = restrictions ? restrictions.x : null;".  Then there's only one place to maintain the property list, and r.clone() becomes an alias for "new Restrictions(r)".